mimic.py

```diff
@@ -0,0 +1,179 @@
+class Node:
+    """
+    Represents a node in MyList object
+    """
+    def __init__(self, data, next_node=None):
+        self.next_node = next_node
+        self.data = data
+
+
+class MyList:
+    """
+    MyList objects have all the methods as list objects
+    """
+
+    def __init__(self):
+        self.length = 0
+        self.first_node = None
+
+    def __len__(self):
+        return self.length
+
+    def __iter__(self):
+        return NodeIterator(self)
+
+    def __getitem__(self, key: int):
+        return self._get_node(key).data
+
+    def __add__(self, other):
+        """
+        :param other: Iterable object to concatenate
+        :return: A new MyList object that includes self and other concatenated
+        """
+        extended = self.copy()
+        extended.extend(other)
+        return extended
+
+    def _get_node(self, key: int) -> Node:
+        """
+        Works like MyList[key] but return the node instead of the value
+
+        :param key: The index in list of the requested Node
+        :return: Node in the position of key
+        """
+        if key >= len(self):
+            raise IndexError('list index is out of range')
+        current = self.first_node
+        i = 0
+        while current:
```

Comment:
Why not use `NodeIterator`?

Reply:
Because NodeIterator return the value but we need the node itself.
(Node has 2 attributes: value and next)